localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -37,6 +36,14 @@ def get_sqs_client(function_version, client_config=None):
     ).sqs
 
 
+# TODO: remove once DLQ handling is refactored following the removal of the legacy lambda provider
```

Comment:
The DLQ utilility should be reworked post v3 but I tried to keep this removal PR minimal and avoid functional changes.

Reply:
good choice :grin: